mintpy/dem_gsi.py

```diff
@@ -175,78 +174,44 @@ def write_vrt_file(meta, fname):
            xs=meta.lon_step, 
            y0=meta.north,
            ys=meta.lat_step,
-           f=os.path.basename(meta.file_path),
+           f=meta.file_path,
            lo=2*meta.width)
 
     # write to vrt file
-    vrt_file = fname + '.vrt'
-    with open(vrt_file, 'w') as f:
+    with open(out_file, 'w') as f:
         f.write(vrt_str)
-    print('write {}'.format(vrt_file))
-
-    return vrt_file
-
-
-def write_isce_metadata(meta, fname):
-    """
-    Write metadata files in ISCE format (.vrt and .xml files)
-    """
-    import isce
-    import isceobj
-
-    # create isce object for xml file
-    img = isceobj.createDemImage()
-    img.setFilename(os.path.abspath(fname))
-    img.setWidth(meta.width)
-    img.setLength(meta.length)
-    img.setAccessMode('READ')
-    img.bands = 1
-    img.dataType = 'SHORT'
-    img.scheme = 'BIP'
-    img.reference = 'WGS84'
-
-    img.firstLatitude  = meta.north + meta.lat_step / 2.
-    img.firstLongitude = meta.west + meta.lon_step / 2.
-    img.deltaLatitude  = meta.lat_step
-    img.deltaLongitude = meta.lon_step
-
-    # write to xml file
-    xml_file = fname + '.xml'
-    img.dump(xml_file)
-
-    return xml_file
+    print('write {}'.format(out_file))
+    return out_file
 
 
 def add_reference_datum(xml_file):
-    """
-    Example of modifying an existing XML file
-    """
-
     import xml.etree.ElementTree as ET
     from xml.dom import minidom
     print('add <reference> info to xml file: {}'.format(os.path.basename(xml_file)))
+    tree = ET.parse(xml_file)
+    root = tree.getroot()
 
     # get property element for reference
-    ref = ET.Element("property", attrib={'name': 'reference'})
+    ref = ET.Element("property")
+    ref.attrib = {'name': 'reference'}
     
-    val = ET.SubElement(ref, "value")
-    val.text = "WGS84"
+    elm1 = ET.Element("value")
+    elm1.text = "WGS84"
+    ref.append(elm1)
     
-    doc = ET.SubElement(ref, "doc")
-    doc.text = "Geodetic datum"
-
-    # pretty xml
-    ref_str = minidom.parseString(ET.tostring(ref)).toprettyxml(indent="    ")
-    ref = ET.fromstring(ref_str)
+    elm1 = ET.Element("doc")
+    elm1.text = "Geodetic datum"
+    ref.append(elm1)
+    
+    ref = ET.fromstring(minidom.parseString(ET.tostring(ref)).toprettyxml(indent="    "))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Using xml.dom.minidom.parseString to parse untrusted XML data is known to be vulnerable to XML attacks. Replace xml.dom.minidom.parseString with its defusedxml equivalent function or make sure defusedxml.defuse_stdlib() is called](https://app.codacy.com/manual/insarlab/MintPy/pullRequest?prid=5671549)